glambie/data/data_catalogue_helpers.py

```diff
@@ -0,0 +1,75 @@
+from glambie.data.data_catalogue import DataCatalogue
+import pandas as pd
+import numpy as np
+from glambie.util.timeseries_combination_helpers import calibrate_timeseries_with_trends, combine_calibrated_timeseries
+from glambie.data.timeseries import TimeseriesData, Timeseries
+import copy
+import warnings
+
+
+def calibrate_timeseries_with_trends_catalogue(catalogue_with_trends: DataCatalogue,
+                                               calibration_timeseries: Timeseries) -> DataCatalogue:
+    """
+    Calibrates a Timeseries of higher resolution varibility with a DataCatalogue of Timeseries of longer-term trends.
+
+    Parameters
+    ----------
+    catalogue_with_trends : DataCatalogue
+        DataCatalogue containing a set of long-term trends.
+    calibration_timeseries : Timeseries
+        Timeseries used for calibration. Should be higher resolution and homogenized.
+
+    Returns
+    -------
+    DataCatalogue
+        Containing all the calibrated Timeseries. Contains the same length of datasets as catalogue_with_trends.
+    """
+
+    if not catalogue_with_trends.datasets_are_same_unit():
+        warnings.warn("Warning: Trends within catalogue are not all the same unit. This might lead to false results.")
+
+    # calibrate annual trends with longterm trend
+    calibrated_series = []
+    for ds in catalogue_with_trends.datasets:
+        # 1) calibrate timeseries
+        calibrated_s, dist_mat = calibrate_timeseries_with_trends(ds.data.as_dataframe(),
+                                                                  calibration_timeseries.data.as_dataframe())
+        # 2) caliculate mean calibration timeseries from all the different curves
+        mean_calibrated_ts = combine_calibrated_timeseries(calibrated_s, dist_mat, p_value=0)
+        df_mean_calibrated = pd.DataFrame({"start_dates": calibration_timeseries.data
+                                           .as_dataframe().start_dates,
+                                           "end_dates": calibration_timeseries.data
+                                           .as_dataframe().end_dates, "changes": mean_calibrated_ts})
+        df_mean_calibrated_na_removed = df_mean_calibrated[~df_mean_calibrated["changes"].isna()]
+
+        # CALCULATE UNCERTAINTIES
+        # The uncertainty of a calibrated time series is calculated by combining the uncertainties of the anomalies
+        # and of the long-term trend
+        df_trends = ds.data.as_dataframe()
+        trend_errors = df_trends.errors  # remove na lines
```

Comment:
the comment here should be on a new line, I think.

Reply:
done